src/components/molecules/SecretPasswordForm/SecretPasswordForm.jsx

```diff
@@ -12,36 +13,38 @@ const SecretPasswordForm = ({ buttonText = "Join the party" }) => {
   const history = useHistory();
   const venueId = useVenueId();
 
-  const [invalidPassword, setInvalidPassword] = useState();
   const [error, setError] = useState();
   const [password, setPassword] = useState();
   const [message, setMessage] = useState();
 
   function passwordChanged(e) {
     setPassword(e.target.value);
-    setInvalidPassword(false);
+    setMessage("");
     setError(false);
   }
 
   const passwordSubmitted = useCallback(
     async (e) => {
       e.preventDefault();
-      setInvalidPassword(false);
       setMessage("Checking password...");
 
-      try {
-        const result = await checkAccess({
-          venueId,
-          password,
+      await checkAccess({
+        venueId,
+        password,
+      })
+        .then((result) => {
+          if (isTruthy(result?.data?.token)) {
+            setLocalStorageToken(venueId, result.data.token);
+            history.push(venueEntranceUrl(venueId));
+          } else {
+            setMessage(`Wrong password!`);
```

Comment:
[nit]

```suggestion
            setMessage('Wrong password!');
```

Reply:
[NB] I believe the `.then()` block will only run if the function doesn't return an 'error condition', so will this part ever run? I believe any 'http errors' returned should trigger the `.catch()` part instead?

Reply:
When the token is doesn't exist this still succeeds but returns `token: null` as result. That's why this check is here.

Reply:
That makes sense.

[NB, perhaps as a future backlog issue] It could be useful to have it return a proper http error in future.